api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -252,5 +255,41 @@ else if (!isObsGroup) {
 			}
 		}
 	}
+
+	/**
+	 * This method validates Obs' concept values.
+	 * <ol>
+	 *     <li>Validates if high absolute and low absolute are within the valid range</li>
+	 *     <li>Validates if patient's age is within the valid range</li>
+	 * <ol/>
+	 * in the {@link ConceptReferenceRange}.
+	 *
+	 * @param obs Observation to validate
+	 * @param errors Errors to record validation issues
+	 */
+	private void validateConceptReferenceRange(Obs obs, Errors errors) {
+		Concept concept = obs.getConcept();
+
+		// if there is a concept perform validation tests specific to the concept datatype
+		if (concept != null && concept.getDatatype() != null) {
```

Comment:
What i would do is, if the concept is numeric, i fetch its corresponding concept reference ranges for validation.

Reply:
In other words we are only validating numeric concepts right? (including criteria evaluation)

Reply:
Yes it applies to only numeric concepts.